wordpress-mobile/WordPress-iOS
# Remove the .showSignupMethod segue and replace with programmatic nav

Fixes # n/a

This PR removes the segue that sends a user to the "Magic Link sent!" screen. It now navigates the user programmatically.

Ref.: https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/225

### To test:
1. Check out this branch
2. `rake dependencies`
3. Build and run
4. Navigate to each of these flows:

#### Flow 1, Login
1. Select the Log in button
2. Select Continue with WordPress.com
3. Tap away from the buttons view to dismiss the buttons 
4. Verify it dismissed (no animation errors)
5. Verify you are navigated to the Email login screen

#### Flow 2, Signup
1. Select the Sign up for WordPress.com button
2. Tap away from the buttons view to dismiss the buttons
3. Verify it dismissed (and there are no animation errors)
4. Select the Sign up with Email button
5. Verify you are navigated to the Email signup screen

### Do not merge until:
0. This PR is merged: https://github.com/wordpress-mobile/WordPress-iOS/pull/13020 (Auth now has a dependency update for WordPressKit 4.7.0-beta.1)
1. The Authenticator PR has been merged: https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/225
2. A new Authenticator release has been created 
3. The release is published on Cocoapods trunk 
4. This PR points to the new release  

PR submission checklist:
